pkg/model/message.go

```diff
@@ -14,6 +14,44 @@ type Messages struct {
 	Original bool
 }
 
+type MessagesSlice []Messages
+
+// GetCopy() returns a deep copy of MessagesSlice.
+func (m MessagesSlice) GetCopy() MessagesSlice {
+	if m == nil {
```

Comment:
This is not a correct expression for a slice for your use case. Which one is?

Reply:
Also, the receiver is passed by a value, not a pointer, `m == nil` is never true.

Reply:
len() -> if empty slice is provided, new empty slice is returned, but I guess this check can be skipped.